aten/src/ATen/native/nested/cuda/NestedTensorTransformerFunctions.cpp

```diff
@@ -474,84 +420,117 @@ std::tuple<Tensor, Tensor> _scaled_dot_product_efficient_attention_nestedtensor_
       {Nnz_kv, num_heads, head_dim_v},
       {nnz_v_stride, head_v_stride, head_dim_stride},
       value_impl->get_storage_offsets()[0]);
-  std::tuple<Tensor, Tensor> attention_and_logsumexp=
-      at::_efficient_attention_forward(
-          query_buffer_reshaped.unsqueeze(0),
-          key_buffer_reshaped.unsqueeze(0),
-          value_buffer_reshaped.unsqueeze(0),
-          cumulative_sequence_length_q,
-          cumulative_sequence_length_k,
-          max_seqlen_batch_q,
-          compute_log_sumexp,
-          is_causal);
-  // Reshape output to convert nnz to batch_size and seq_len
-  Tensor attention = std::get<0>(attention_and_logsumexp);
-  auto attention_size = get_nested_size_tensor(q_t).clone();
+
+  auto output_shape = get_nested_size_tensor(q_t).clone();
   if (head_dim_v != head_dim_qk) {
-    attention_size.select(1, -1).fill_(head_dim_v);
+    output_shape.select(1, -1).fill_(head_dim_v);
   }
-  attention =
-      wrap_buffer(attention.view(-1), attention_size)
-          .transpose(1, 2);
-  return std::tie(attention, std::get<1>(attention_and_logsumexp));
+
+  return std::make_tuple(
+      query_buffer_reshaped,
+      key_buffer_reshaped,
+      value_buffer_reshaped,
+      cumulative_sequence_length_q,
+      cumulative_sequence_length_kv,
+      max_seqlen_batch_q,
+      max_seqlen_batch_kv,
+      output_shape);
```

Comment:
```suggestion
    return std::make_tuple(
      std::move(query_buffer_reshaped),
      std::move(key_buffer_reshaped),
      std::move(value_buffer_reshaped),
      cumulative_sequence_length_q,
      cumulative_sequence_length_kv,
      max_seqlen_batch_q,
      max_seqlen_batch_kv,
      std::move(output_shape));

Reply:
A few questions I have about these. If add the standard move clang-tidy will warn me with `Moving a temporary object prevents copy elision (fix available)`. I am assuming that these objects will get moved when building PyTorch with optimizations turned on.

Second question aren't Tensors trivially copyable anyways so we don't need to be moving?